diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/BlockStructureBraces.kt

```diff
@@ -104,22 +109,27 @@ class BlockStructureBraces : Rule("block-structure") {
 
     private fun checkLoop(node: ASTNode, configuration: BlockStructureBracesConfiguration) {
         node.findChildByType(BODY)?.takeIf { body -> body.hasChildOfType(BLOCK) }?.let {
-            checkOpenBraceOnSameLine(node, BODY, configuration)
-            checkCloseBrace(it.findChildByType(BLOCK)!!, configuration)
-            if (node.elementType == DO_WHILE){
-                val allMiddleNode = listOf(node.findChildByType(BODY)!!.treeNext)
-                checkMidBrace(allMiddleNode, node, WHILE_KEYWORD)
+            if (!it.findChildByType(BLOCK).isBlockEmpty()) {
+                checkOpenBraceOnSameLine(node, BODY, configuration)
+                checkCloseBrace(it.findChildByType(BLOCK)!!, configuration)
+                if (node.elementType == DO_WHILE) {
+                    val allMiddleNode = listOf(node.findChildByType(BODY)!!.treeNext)
+                    checkMidBrace(allMiddleNode, node, WHILE_KEYWORD)
+                }
             }
         }
     }
 
     private fun checkWhen(node: ASTNode, configuration: BlockStructureBracesConfiguration) {
-        checkOpenBraceOnSameLine(node, LBRACE, configuration)
-        checkCloseBrace(node, configuration)
+        val childrenAfterLBrace =  node.getChildren(null).toList().run { subList(indexOfFirst { it.elementType == LBRACE }, size) }
```

Comment:
In which case will `LBRACE` be not the first child? Please add a comment explaining this, it seems not obvious.

Reply:
WHEN expression doesn't contain BLOCK element and LBRECE isn't the first child, so we should to find it.